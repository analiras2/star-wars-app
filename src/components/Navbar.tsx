import React, { PureComponent } from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import colors from '../res/colors';
import strings from '../res/strings';
import SWText from './SWText';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
    height: 52,
    alignItems: 'center',
    borderBottomColor: colors.divider,
    borderBottomWidth: 1,
  },
  icon: { marginLeft: 16, width: 52, height: 52, justifyContent: 'center' },
});

interface Props {
  title?: string;
  onBackPress?: () => void;
}
export default class Navbar extends PureComponent<Props> {
  public render() {
    const { title = strings.appName, onBackPress } = this.props;

    return (
      <View style={styles.container}>
        {onBackPress && (
          <TouchableOpacity
            style={styles.icon}
            activeOpacity={0.6}
            onPress={onBackPress}
          >
            <Icon name="md-arrow-back" color={colors.secondary} size={24} />
          </TouchableOpacity>
        )}
        <SWText
          flex={1}
          title={title.toUpperCase()}
          textAlign="center"
          color={colors.accent}
          medium
          bold
          spacing={2}
        />
        {onBackPress && <View style={styles.icon} />}
      </View>
    );
  }
}
