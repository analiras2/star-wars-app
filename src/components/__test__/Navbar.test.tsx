import {shallow} from 'enzyme';
import React from 'react';
import {render} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import Navbar from "../Navbar";
import {TouchableOpacity} from "react-native";

describe('Navbar', () => {
    const props = { title: 'Star Wars App', onBackPress: () => {} };
    const wrapper = shallow(<Navbar />);
    const wrapperWithProps = shallow(<Navbar {...props} />);
    const { getByText } = render(<Navbar />);

    describe('validating Navbar children', () => {
        it('should render a Star Wars App', () => {
            expect(getByText(/Star Wars App/i)).toBeDefined();
        });
    });

    describe('rendering Components', () => {
        describe('Without props', () => {
            it('should render a <View />', () => {
                expect(wrapper.find('View')).toHaveLength(1);
            });

            it('should render a <SWText />', () => {
                expect(wrapper.find('SWText')).toHaveLength(1);
            });

            it('should render a <SWText />', () => {
                expect(wrapper.contains(<TouchableOpacity />)).toBe(false)
            });
        });

        describe('With props', () => {
            it('should render a <TouchableOpacity />', () => {
                expect(wrapperWithProps.find('TouchableOpacity')).toHaveLength(1);
            });

            it('should render a <View />', () => {
                expect(wrapperWithProps.find('View')).toHaveLength(2);
            });
        });
    });

    it('renders SWText correctly', () => {
        const tree = renderer.create(<Navbar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
