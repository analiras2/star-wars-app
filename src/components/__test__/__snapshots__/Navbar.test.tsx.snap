// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders SWText correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "rgba(255,232,31, 0.5)",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 52,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "auto",
        "color": "#FFE81F",
        "flex": 1,
        "fontFamily": "Exo-Bold",
        "fontSize": 18,
        "letterSpacing": 2,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "center",
      }
    }
  >
    STAR WARS APP
  </Text>
</View>
`;
